File: packages/typescript/src/tools/SwitchToPreviousTabTool.ts
import { BaseDriver } from "../drivers/BaseDriver.ts";
import { BaseTool } from "./BaseTool.ts";

export class SwitchToPreviousTabTool extends BaseTool {
  static description = `Switch to the previous browser tab/window.`;

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.switchToPreviousTab();
  }
}

File: packages/java/src/main/java/ai/alumnium/tool/SwitchToPreviousTabTool.java
package ai.alumnium.tool;

import ai.alumnium.driver.BaseDriver;
import ai.alumnium.tool.annotation.ToolDescription;

@ToolDescription("Switch to the previous browser tab/window.")
public record SwitchToPreviousTabTool() implements BaseTool {
  @Override
  public void invoke(BaseDriver driver) {
    driver.switchToPreviousTab();
  }
}
